ooni/deck/store.py

```diff
@@ -0,0 +1,193 @@
+        yield self.create(country_code)
+
+    def _update_cache(self):
+        descs = self.path.child("descriptors")
+        if not descs.exists():
+            self._cache = {}
+            return
+
+        for fn in descs.listdir():
+            with descs.child(fn).open("r") as in_fh:
+                input_desc = json.load(in_fh)
+                self._cache[input_desc.pop("id")] = input_desc
+        self._cache_stale = False
+        return
+
+    def list(self):
+        if self._cache_stale:
+            self._update_cache()
+        return deepcopy(self._cache)
+
+    def get(self, input_id):
+        if self._cache_stale:
+            self._update_cache()
+        try:
+            input_desc = self._cache[input_id]
+        except KeyError:
+            raise InputNotFound(input_id)
+        return input_desc
+
+    def getContent(self, input_id):
+        input_desc = self.get(input_id)
+        with open(input_desc["filepath"]) as fh:
+            return fh.read()
+
+
+class DeckStore(object):
+    def __init__(self):
+        self.enabled_directory = FilePath(config.decks_enabled_directory)
+        self.available_directory = FilePath(config.decks_available_directory)
+        self._cache = {}
+        self._cache_stale = True
+
+    def _list(self):
+        if self._cache_stale:
+            self._update_cache()
+        for deck_id, deck in self._cache.iteritems():
+            yield (deck_id, deck)
+
+    def list(self):
+        decks = []
+        for deck_id, deck in self._list():
+            decks.append((deck_id, deck))
+        return decks
+
+    def list_enabled(self):
+        decks = []
+        for deck_id, deck in self._list():
+            if not self.is_enabled(deck_id):
+                continue
+            decks.append((deck_id, deck))
+        return decks
+
+    def is_enabled(self, deck_id):
+        return self.enabled_directory.child(deck_id + '.yaml').exists()
+
+    def enable(self, deck_id):
+        deck_path = self.available_directory.child(deck_id + '.yaml')
+        if not deck_path.exists():
+            raise DeckNotFound(deck_id)
+        deck_enabled_path = self.enabled_directory.child(deck_id + '.yaml')
+        try:
+            deck_path.linkTo(deck_enabled_path)
+        except OSError as ose:
+            if ose.errno != errno.EEXIST:
+                raise
+
+    def disable(self, deck_id):
+        deck_enabled_path = self.enabled_directory.child(deck_id + '.yaml')
+        if not deck_enabled_path.exists():
+            raise DeckNotFound(deck_id)
+        deck_enabled_path.remove()
+
+    def _update_cache(self):
+        for deck_path in self.available_directory.listdir():
+            if not deck_path.endswith('.yaml'):
+                continue
+            deck_id = deck_path[:-1*len('.yaml')]
+            deck = NGDeck(
+                deck_path=self.available_directory.child(deck_path).path
+            )
+            self._cache[deck_id] = deck
```

Comment:
Uhm, shouldn't this function set `_cache_stale` to `False`?

Also, I do not understand why the function updating the cache does not restart from an empty cache (i.e. from `self._cache = {}`)